### PR TITLE
Testing: Mock the google maps geocoding API in Jasmine

### PR DESCRIPTION
**Pull request by nstjean**
#233 

- Removes all reliance on the google API during tests (does not require a key or api call)
- Using mocked data created 3 tests in geocoding_spec.js for the functions getPlacenameFromCoordinates and geocodeStringAndPan
- Removed the "Can't find variable: google" warning in the other test spec files that would happen when the blurredLocation object was initialized
- I will be trying to move the mock object and the response object to a helper file, but I'm having some issues getting jasmine to recognize any helper files.

Be aware that if the google maps api ever changes we will also have to change our mock object in testing.

I suspect I will have to do a similar mock/stub in LBLD and possibly LEL to correct the same error with the blurredLocation object in testing.

### CODE REVIEW

**Comment by nstjean** [comment]
I figured out how to move it to a helper file! The existing `support/jasmine.json` file was NOT being checked by the script. I had to add a helper directory in `Gruntfile.js`!

File: src/ui/Interface.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ module.exports = function Interface (options) {
   options.onDrag = options.onDrag || function onDrag() {
     function onPlacenameReturned(result) {
       result = (result) ? result : options.placenameDisplayOnError; // this makes jasmine pass, other formats don't
-      $("#"+options.placenameDisplayId).val(result);
+      $("#"+options.placenameDisplayId).val(result.trim());
```
Comment (sagarpreet-chadha) [inline src/ui/Interface.js]: Awesome!

**Comment by nstjean** [comment]
I have added error checks in `Geocoding.js` so that the functions will only execute if there is a google object - either a real one or a mock one. It was a bit tricky getting it set up so that it didn't mess up the mock object I just created in the tests, but I think I got it! I tested this on gh-pages, you can see that "the search for a place name" and "placename" display still work properly : https://publiclab.github.io/leaflet-blurred-location/examples/index.html

If this works properly it SHOULD mean that my fixes in LBLD and LEL are no longer needed! This should mean no warnings or errors even when there's no google api key given.

**Comment by jywarren** [comment]
Awesome!

**Comment by nstjean** [comment]
@jywarren This is all set to merge! And once merged can you publish to npm?

**Comment by jywarren** [comment]
Sorry I missed this! Doing it now, thank you!!!

**Comment by jywarren** [comment]
Done!

**Pull request merged**